### PR TITLE
Chunk session recordings

### PR DESCRIPTION
**Pull request by mariusandra**
## Changes

- Chunks session recordings into blocks of 512KB when ingesting
- Combines the blocks when reading back from the database

## Checklist

- [ ] All querysets/queries filter by Organization, by Team, and by User
- [ ] Django backend tests
- [ ] Jest frontend tests
- [ ] Cypress end-to-end tests

### CODE REVIEW

**Comment by mariusandra** [comment]
I seem to have gotten flaky tests. If they'd pass, this PR would be ready to be reviewed.

I'm especially looking for scrutiny of the shape of the [chunk objects](https://github.com/PostHog/posthog/pull/3566/files#diff-6776f6bb00ceae244fa51c7a088584917bd4e4baf9bcb1cfdcb1f63de007cf9aR113-R121).

Before merging any session recording chunking, and to keep the unknown-to-me dragons away, I think we also need @macobo 's blessing on this approach.

**Review by Twixes — COMMENTED** [review]
Code looks good to me, though `TestClickhouseSessionsList` and `TestClickhouseSessions` seem to be exceptionally flaky for this PR… Maybe there's a condition when merging doesn't occur properly, or it can be a false alarm.
Wanted to test playing session recording, but unfortunately when trying out this PR's Postgres review app, no session in the list has "Play recording" available. :(

**Comment by mariusandra** [comment]
To be fair, that test seems flaky everywhere now :/ 

Also in [This PR](https://github.com/PostHog/posthog/pull/3570#pullrequestreview-604131328) for example:

image

image

Also in this PR: https://github.com/PostHog/posthog/pull/3579

**Comment by macobo** [comment]
I did not yet dig deep in here, however the missing piece here is compressing the json payload before chunking. With the data we're working with gzip can make a difference of 10-100x of the data.

File: posthog/queries/sessions/session_recording.py [hunk 1]
```diff
@@ -59,10 +61,54 @@ def query_recording_snapshots(
...
+            elif snapshot_or_collector.get("snapshot"):
+                snapshots.append(snapshot_or_collector["data"])
+
+        return snapshots
+
     def run(self, team: Team, session_recording_id: str, *args, **kwargs) -> Dict[str, Any]:
         from posthog.api.person import PersonSerializer
 
-        distinct_id, start_time, snapshots = self.query_recording_snapshots(team, session_recording_id)
+        distinct_id, start_time, unmerged_snapshots = self.query_recording_snapshots(team, session_recording_id)
```
Comment (macobo) [inline posthog/queries/sessions/session_recording.py:110]: Will SESSIONS_IN_RANGE_QUERY above still continue working with this setup? From a cursory glance at the code it seems not.

Basically there will always be sessions with incomplete data - e.g. person leaves the page before posthog.js finishes sending the full payload event or the larger payload gets blocked. We should not show sessions where there's no full snapshot event which the below line accomplishes:

`COUNT(*) FILTER(where snapshot_data->>'type' = '2') as full_snapshots`

**Comment by macobo** [comment]
Superceded by another PR that is already live on cloud and coming up with the enxt release!

**Pull request closed without merging**